File: javascript/anagram/anagram.js
// This is only a SKELETON file for the 'Anagram' exercise. It's been provided as a
// convenience to get you started writing code faster.
export class Anagram {
  constructor(word) {
    this.word = word.toUpperCase()    
  }

  matches(array) {
    const NUEVO = []
    let x = array.map( i => 
      (i.toUpperCase().split('').sort().join(''))
    )
    x.map((i, j) => {
      if(i === this.word.split('').sort().join('') && this.word !== array[j].toUpperCase()){
        NUEVO.push(array[j])
      }
    })
    return NUEVO
  }
}

File: ruby/anagram/anagram.rb
# frozen_string_literal: true

# Exercism Anagram
class Anagram
  attr_reader :word

  def initialize(word)
    @word = word
  end

  def match(anagrams)
    anagrams.select do |element|
      element.upcase != word.upcase && modify_word(element) == modify_word(word)
    end
  end

  private

  def modify_word(str)
    str.upcase.split('').sort.reverse.join
  end
end
